analysis/analysis.go

```diff
@@ -27,15 +27,25 @@ import (
 
 	"github.com/googleapis/gnostic-grpc/incompatibility"
 	"github.com/googleapis/gnostic-grpc/utils"
-	"google.golang.org/protobuf/encoding/prototext"
 )
 
 // main function for aggreation tool
 func main() {
 	if len(os.Args) != 2 {
 		exitIfError(errors.New("argument should be a path to a directory"))
 	}
-	generateAnalysis(os.Args[1])
+	dir := os.Args[1]
+	analysis := generateAnalysis(dir)
+	writeAnalysis(dir, analysis)
+}
+
+func writeAnalysis(analysisName string, analysis *incompatibility.ApiSetIncompatibility) {
+	pbMessage, msgErr := utils.ProtoTextBytes(analysis)
+	exitIfError(msgErr)
```

Comment:
This isn't very idiomatic of go.  The more common approach is to return an error.  At the top-level main function, you would translate that into an exit. 

This does slightly obfuscate that the application could exit for example when this method is called from main. 
I realize gnostic uses this pattern.

Thoughts?

Reply:
error now handled at main